go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -566,7 +566,14 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 	}
 
 	// CNI depends on the flows from port security, delay setting it until end
-	lsp.PortSecurity = addresses
+	if !bnc.NetInfo.DisablePortSecurity() {
+		lsp.PortSecurity = addresses
+	} else {
+		// if we want to allow traffic from other workloads through, we must add
```

Comment:
I don't quite understand this change. It is also different from downstream implementation of skipPortSecurity. We simply leave lsp.PortSecurity to empty without reset lsp.Addresses.

from ovn-nb manpage, that is sufficient?

```
       port_security: set of strings
              This column controls the addresses from which the host  attached
              to  the  logical  port (``the host’’) is allowed to send packets
              and to which it is allowed to receive packets. If this column is
              empty, all addresses are permitted.
```

Reply:
From the OVN man page unknown addresses explanation:
>This indicates that the logical port has an unknown
  set of Ethernet addresses. When an OVN logical
  switch processes a unicast Ethernet frame whose
  destination MAC address is not in any logical
  port’s addresses column, it delivers it to the port
  (or ports) whose addresses columns include unknown.

Remember the objective of this PR is to allow `nested virtualization`. Thus, we need traffic to be able to be sent to the workloads "behind" the logical port where we are setting the `unknown` address attribute.

Without it, it won't work.

> We simply leave lsp.PortSecurity to empty without reset lsp.Addresses.

And we're not resetting the addresses - we're adding the unknown address to it. That is preferable, as per this thread: https://github.com/ovn-org/ovn-kubernetes/pull/4377#discussion_r1609794722

Reply:
Port-security and "unknown" address are independent features. They can be used in different scenarios and can also be used together to achieve certain requirement. For example, for nested virtualization that is routed from the parent LSP, the LSP receives traffic destined to the LSP's MAC address, in which case we don't need "unknown". But if the nested workload is L2 bridged, it requires that packets destined to the nested workload's MAC address be delivered to the parent LSP, then "unknown" address is required to be added to the parent LSP. In this case, assuming that there are many such LSPs on the logical switch, to avoid every such packet being flooded to all the LSPs (since they all have "unknown" address), we should also set options:force_fdb_lookup to true.

For this patch, since port-security and "unknow" address are independent features, I think it is better to have a separate flag to indicate the "unknown" address. The DisablePortSecurity flag should only affect the port-security setting.

Reply:
Thanks for the elaborate comment @hzhou8 .

We personally only care about L2 bridged, hence this minimal approach.

I do agree these are actually 2 separate features, which should be controlled by two separate knobs. Really good feedback.

Any suggestion on what to call the `unknown` address knob ? ... 
@hzhou8

Reply:
@maiqueb Probably we can call it EnableL2Unknown? When this knob is turned on, we should add "unknown" to the addresses, and in my opinion we should also implicitly set the options:force_fdb_lookup to true, unless we want yet another knob. I'd like to hear opinions from other folks.

Reply:
The less knobs the better IMHO :)

Reply:
@hzhou8 o/ 

I can't seem to find the `` attribute in the [man](https://man7.org/linux/man-pages/man5/ovn-nb.5.html) page ...

The only similar attribute I could is `localnet_learn_fdb`. 
```
       options : localnet_learn_fdb: optional string, either true or false
       
       Optional. Allows localnet port to learn MACs and store
       them in FDB table if set to true. The default value is
       false.
```

Reply:
@hzhou8 another thing I just thought ... assuming we add the two knobs (`disablePortSecurity` and `EnableL2Unknown`), what would the following scenario mean ? 
- enabled port security and enable L2 unknown 

Would that work ?... IIUC, all the traffic meant for the "unknown" addresses would be dropped. Thus, I think that setting is pointless.

This makes me think it could probably be preferable to have a single knob - but not a boolean. Something that had three different values: port security enabled ; port security disabled; port security disabled **and** allow unknown addresses. 

Does that make sense to you ?

Reply:
Hi @maiqueb , I agree that enabled port security and enable L2 unknown doesn't make much sense, so in my opinion by enabling L2 unknown we can implicitly disable port security (and ignoring the port security config), same as what @jcaamano suggested.

Reply:
Thank you. 

Once I can find some time to work on this, I'll re-design the API accordingly.